pkg/provision/engine.go

```diff
@@ -109,20 +136,31 @@ func (e *Engine) Run(ctx context.Context) error {
 				slog.Info().Msg("start decommissioning reservation")
 				if err := e.decommission(ctx, reservation); err != nil {
 					log.Error().Err(err).Msgf("failed to decommission reservation %s", reservation.ID)
+					canCleanup = true
```

Comment:
so if I understand, as soon as one reservation is processed, we allow the clean up to run ?
I think this approach is still a bit dangerous, cause in the case of a reboot, nothing ensure you that all the workloads that should be provisioned actually are. So to me we need to have a way to explicitly know that we did process all the workloads before we allow cleanup to run.

Reply:
How can we know provisiond is done with redeploying after a reboot?

Reply:
I don't think "canCleanup" is needed at all. IF you are in the select statement only once case can run at a time. I don't see why you still have this after chaning to use channels

Reply:
> How can we know provisiond is done with redeploying after a reboot?

One way to do it would be to be sure that last reservation processed is also the most recent one existing on the explorer.

Reply:
> I don't think "canCleanup" is needed at all. IF you are in the select statement only once case can run at a time. I don't see why you still have this after chaning to use channels

Channel does solve the problem I'm rising. You process one reservation at a time. In the case the provisioning of all the required reservations would take more then 5 minutes. We could start deleting primitives that actually should be kept.